src/napi/module.cpp

```diff
@@ -55,8 +56,16 @@ static napi_value setAPIVersion(napi_env env, napi_callback_info info) {
 
   int32_t apiVersion;
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[0], &apiVersion));
-
-  FDB_OK_OR_RETURN_NULL(env, fdb_select_api_version(apiVersion));
+  
+  if (previousApiVersion != 0) {
+    if (apiVersion != previousApiVersion) {
+      napi_throw_error(env, NULL, "foundationdb already initialized with another API version");
```

Comment:
This is missing the correct foundationdb error code. And by supplying that error code, the string shouldn't be necessary.

I think it'll be easier to just call `fdb_select_api_version` if the version doesn't match, and let that function error back to the user in the standard way.